packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -12,325 +12,271 @@ import 'package:flutter/services.dart';
+            const SizedBox(height: 16.0),
+            Container(
+              height: kMinInteractiveDimension * 2,
+              child: Row(
+                children: <Widget>[
+                  if (!use24HourDials && timeOfDayFormat == TimeOfDayFormat.a_space_h_colon_mm) ...<Widget>[
+                    _DayPeriodControl(
+                      selectedTime: selectedTime,
+                      orientation: orientation,
+                      onChanged: onChanged,
+                    ),
+                    const SizedBox(width: 12.0),
+                  ],
+                  Expanded(child: _HourControl(fragmentContext: fragmentContext)),
+                  _StringFragment(timeOfDayFormat: timeOfDayFormat),
+                  Expanded(child: _MinuteControl(fragmentContext: fragmentContext)),
+                  if (!use24HourDials && timeOfDayFormat != TimeOfDayFormat.a_space_h_colon_mm) ...<Widget>[
+                    const SizedBox(width: 12.0),
+                    _DayPeriodControl(
+                      selectedTime: selectedTime,
+                      orientation: orientation,
+                      onChanged: onChanged,
+                    ),
+                  ]
+                ],
               ),
             ),
+          ],
+        );
+        break;
+      case Orientation.landscape:
+        width = _kTimePickerHeaderLandscapeWidth;
+        padding = const EdgeInsets.symmetric(horizontal: 24.0);
+        controls = Expanded(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              if (!use24HourDials && timeOfDayFormat == TimeOfDayFormat.a_space_h_colon_mm)
+                _DayPeriodControl(
+                  selectedTime: selectedTime,
+                  orientation: orientation,
+                  onChanged: onChanged,
+                ),
+              Container(
+                height: kMinInteractiveDimension * 2,
+                child: Row(
+                  children: <Widget>[
+                    Expanded(child: _HourControl(fragmentContext: fragmentContext)),
+                    _StringFragment(timeOfDayFormat: timeOfDayFormat),
+                    Expanded(child: _MinuteControl(fragmentContext: fragmentContext)),
+                  ],
+                ),
+              ),
+              if (!use24HourDials && timeOfDayFormat != TimeOfDayFormat.a_space_h_colon_mm)
+                _DayPeriodControl(
+                  selectedTime: selectedTime,
+                  orientation: orientation,
+                  onChanged: onChanged,
+                ),
+            ],
           ),
-        ),
+        );
+        break;
+    }
+
+    return Container(
+      width: width,
+      padding: padding,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          const SizedBox(height: 16.0),
+          Text(
+            // TODO(rami-a): localize 'SELECT TIME.'
```

Comment:
This might need a github issue to go with the TODO

Reply:
It's not required and so I'm just going to leave this, I'll be following up with that work